lib/IDE/SyntaxModel.cpp

```diff
@@ -772,6 +774,7 @@ bool ModelASTWalker::walkToDeclPre(Decl *D) {
                                                      AFD->getBodySourceRange());
       SN.NameRange = charSourceRangeFromSourceRange(SM,
                           AFD->getSignatureSourceRange());
+      SN.DocRange = findDocCommentRange(AFD->getAttrs());
```

Comment:
can we have a function called `setDecl(SyntaxStructureNode&,Decl*)`; where it sets both the decl pointer `SN.Dcl` and `SN.DocRange`? so all these added lines can call this function.